### DISCUSSION TITLE
passwall 有办法实现不要让bing重定向到cn.bing吗

### DISCUSSION THREAD

**Discussion by jt8823** [Openwrt-Passwall/openwrt-passwall#2815] [General]
我已经把bing加入到了代理列表,但是访问bing的时候,还是会重定向到cn.bing

**Comment by naoki66** [comment]
这个真是刚需

**Reply by jxr111** [reply]
passwall可以配合smartdns，对于bing.com的域名指定通过墙外dns解析，亲测可以

**Reply by Lonely-Sit** [reply]
请问要如何配置呢？

**Comment by colalan** [comment]
改下usr\share\passwall\rule_update.lua：

27行：

local excluded_domain = {"apple.com","sina.cn","sina.com.cn","baidu.com","byr.cn","jlike.com","weibo.com","zhongsou.com","youdao.com","sogou.com","so.com","soso.com","aliyun.com","taobao.com","jd.com","qq.com","bing.com"}

删除bing.com改为：
local excluded_domain = {"apple.com","sina.cn","sina.com.cn","baidu.com","byr.cn","jlike.com","weibo.com","zhongsou.com","youdao.com","sogou.com","so.com","soso.com","aliyun.com","taobao.com","jd.com","qq.com"}

154行左右以下：

                        if rule_name == "gfwlist" then
                                local domains = {"bing.com"}
                                local gfwlist = io.open(download_file_tmp..k, "r")
                                local decode = api.base64Decode(gfwlist:read("*all"))
                                gfwlist:close()

                                gfwlist = io.open(download_file_tmp..k, "w")
                                gfwlist:write(decode)
                                for i = 1, #domains do
                                        gfwlist:write('\n', domains)
                                end
                                gfwlist:close()
                        end



215行左右以下：

                if rule_type == "domain" then
                        local out = io.open(unsort_file_tmp, "w")
                        for k,v in pairs(domains) do
                                if k ~= "www.bing.com" then
                                        out:write(string.format("%s\n", k))
                                end
                        end
                        out:close()
                end

**Reply by jt8823** [reply]
不对呀,我的rule_update.lua 里面  没有这些内容啊
#!/usr/bin/lua

require 'nixio'
require 'luci.sys'
local luci = luci
local ucic = luci.model.uci.cursor()
local jsonc = require "luci.jsonc"
local name = 'passwall2'
local api = require "luci.passwall2.api"
local arg1 = arg[1]

local reboot = 0
local geoip_update = 0
local geosite_update = 0
local asset_location = ucic:get_first(name, 'global_rules', "v2ray_location_asset", "/usr/share/v2ray/")

-- Custom geo file
local geoip_api = ucic:get_first(name, 'global_rules', "geoip_url", "https://api.github.com/repos/Loyalsoldier/v2ray-rules-dat/releases/latest")
local geosite_api = ucic:get_first(name, 'global_rules', "geosite_url", "https://api.github.com/repos/Loyalsoldier/v2ray-rules-dat/releases/latest")
--
local use_nft = ucic:get(name, "@global_forwarding[0]", "use_nft") or "0"

local log = function(...)
	if arg1 then
		if arg1 == "log" then
			api.log(...)
		elseif arg1 == "print" then
			local result = os.date("%Y-%m-%d %H:%M:%S: ") .. table.concat({...}, " ")
			print(result)
		end
	end
end

-- curl
local function curl(url, file)
	local args = {
		"-skL", "-w %{http_code}", "--retry 3", "--connect-timeout 3"
	}
	if file then
		args[#args + 1] = "-o " .. file
	end
	local return_code, result = api.curl_logic(url, nil, args)
	return tonumber(result)
end

--获取geoip
local function fetch_geoip()
	--请求geoip
	xpcall(function()
		local return_code, content = api.curl_logic(geoip_api)
		local json = jsonc.parse(content)
		if json.tag_name and json.assets then
			for _, v in ipairs(json.assets) do
				if v.name and v.name == "geoip.dat.sha256sum" then
					local sret = curl(v.browser_download_url, "/tmp/geoip.dat.sha256sum")
					if sret == 200 then
						local f = io.open("/tmp/geoip.dat.sha256sum", "r")
						local content = f:read()
						f:close()
						f = io.open("/tmp/geoip.dat.sha256sum", "w")
						f:write(content:gsub("geoip.dat", "/tmp/geoip.dat"), "")
						f:close()

						if nixio.fs.access(asset_location .. "geoip.dat") then
							luci.sys.call(string.format("cp -f %s %s", asset_location .. "geoip.dat", "/tmp/geoip.dat"))
							if luci.sys.call('sha256sum -c /tmp/geoip.dat.sha256sum > /dev/null 2>&1') == 0 then
								log("geoip 版本一致，无需更新。")
								return 1
							end
						end
						for _2, v2 in ipairs(json.assets) do
							if v2.name and v2.name == "geoip.dat" then
								sret = curl(v2.browser_download_url, "/tmp/geoip.dat")
								if luci.sys.call('sha256sum -c /tmp/geoip.dat.sha256sum > /dev/null 2>&1') == 0 then
									luci.sys.call(string.format("mkdir -p %s && cp -f %s %s", asset_location, "/tmp/geoip.dat", asset_location .. "geoip.dat"))
									reboot = 1
									log("geoip 更新成功。")
									return 1
								else
									log("geoip 更新失败，请稍后再试。")
								end
								break
							end
						end
					end
					break
				end
			end
		end
	end,
	function(e)
	end)

	return 0
end

--获取geosite
local function fetch_geosite()
	--请求geosite
	xpcall(function()
		local return_code, content = api.curl_logic(geosite_api)
		local json = jsonc.parse(content)
		if json.tag_name and json.assets then
			for _, v in ipairs(json.assets) do
				if v.name and v.name == "geosite.dat.sha256sum" then
					local sret = curl(v.browser_download_url, "/tmp/geosite.dat.sha256sum")
					if sret == 200 then
						local f = io.open("/tmp/geosite.dat.sha256sum", "r")
						local content = f:read()
						f:close()
						f = io.open("/tmp/geosite.dat.sha256sum", "w")
						f:write(content:gsub("geosite.dat", "/tmp/geosite.dat"), "")
						f:close()

						if nixio.fs.access(asset_location .. "geosite.dat") then
							luci.sys.call(string.format("cp -f %s %s", asset_location .. "geosite.dat", "/tmp/geosite.dat"))
							if luci.sys.call('sha256sum -c /tmp/geosite.dat.sha256sum > /dev/null 2>&1') == 0 then
								log("geosite 版本一致，无需更新。")
								return 1
							end
						end
						for _2, v2 in ipairs(json.assets) do
							if v2.name and v2.name == "geosite.dat" then
								sret = curl(v2.browser_download_url, "/tmp/geosite.dat")
								if luci.sys.call('sha256sum -c /tmp/geosite.dat.sha256sum > /dev/null 2>&1') == 0 then
									luci.sys.call(string.format("mkdir -p %s && cp -f %s %s", asset_location, "/tmp/geosite.dat", asset_location .. "geosite.dat"))
									reboot = 1
									log("geosite 更新成功。")
									return 1
								else
									log("geosite 更新失败，请稍后再试。")
								end
								break
							end
						end
					end
					break
				end
			end
		end
	end,
	function(e)
	end)

	return 0
end

if arg[2] then
	string.gsub(arg[2], '[^' .. "," .. ']+', function(w)
		if w == "geoip" then
			geoip_update = 1
		end
		if w == "geosite" then
			geosite_update = 1
		end
	end)
else
	geoip_update = ucic:get_first(name, 'global_rules', "geoip_update", 1)
	geosite_update = ucic:get_first(name, 'global_rules', "geosite_update", 1)
end
if geoip_update == 0 and geosite_update == 0 then
	os.exit(0)
end

log("开始更新规则...")

if tonumber(geoip_update) == 1 then
	log("geoip 开始更新...")
	local status = fetch_geoip()
	os.remove("/tmp/geoip.dat")
	os.remove("/tmp/geoip.dat.sha256sum")
end

if tonumber(geosite_update) == 1 then
	log("geosite 开始更新...")
	local status = fetch_geosite()
	os.remove("/tmp/geosite.dat")
	os.remove("/tmp/geosite.dat.sha256sum")
end

ucic:set(name, ucic:get_first(name, 'global_rules'), "geoip_update", geoip_update)
ucic:set(name, ucic:get_first(name, 'global_rules'), "geosite_update", geosite_update)
ucic:save(name)
luci.sys.call("uci commit " .. name)

if reboot == 1 then
	log("重启服务，应用新的规则。")
	if use_nft == "1" then
		luci.sys.call("sh /usr/share/" .. name .. "/nftables.sh flush_nftset > /dev/null 2>&1 &")
	else
		luci.sys.call("sh /usr/share/" .. name .. "/iptables.sh flush_ipset > /dev/null 2>&1 &")
	end
end
log("规则更新完毕...")

**Reply by colalan** [reply]
你确认使用的是原版openwrt-passwall吗？你看下原码：https://github.com/xiaorouji/openwrt-passwall/blob/main/luci-app-passwall/root/usr/share/passwall/rule_update.lua

**Reply by colalan** [reply]
啊，你用的是passwall2不是passwall，那个passwall2不知怎么改，它是个简版passwall。

**Reply by jt8823** [reply]
嗯,  我用的passwall2,     谢了!!!

**Comment by sofs2005** [comment]
试试看，不知道能不能把copilot给救活

**Reply by sofs2005** [reply]
好像不太好使，会先跳cn然后再到无cn的，后面就打不开了

该网页无法正常运作cn.bing.com 将您重定向的次数过多。
[尝试删除您的 Cookie](https://support.google.com/chrome?p=rl_error&hl=zh-CN).
ERR_TOO_MANY_REDIRECTS

**Reply by jt8823** [reply]
我现在是把微软的所有域名，添加到规则里面了，  能正常使用。

**Reply by colalan** [reply]
也许是我漏说了一句，改完“usr\share\passwall\rule_update.lua”，还需要到passwall的“规则管理”下点击一下“手功更新”，让规则按改动更新一次。
我这里很正常的，未再发生过重定向，电脑的copilot也好，网页的bing也好，手机的copilot也好，一切正常使用。
也可以试试bypass，它的修改更简单，改下usr\share\bypass\by-gfw，新增代码：“**sed '$a bing.com' |**”

        cat /tmp/gfwlist.txt | sort -u |
                sed 's#!.\+##; s#|##g; s#@##g; s#http:\/\/##; s#https:\/\/##;' |
                sed '/\*/d; /apple\.com/d; /sina\.cn/d; /sina\.com\.cn/d; /baidu\.com/d; /byr\.cn/d; /jlike\.com/d; /weibo\.com/d; /zhongsou\.com/d; /youdao\.com/d; /sogou\.com/d; /so\.com/d; /soso\.com/d; /aliyun\.com/d; /taobao\.com/d; /jd\.com/d; /qq\.com/d' |
                sed '/haosou\.com/d /ip\.cn/d /jike\.com/d /gov\.cn/d /uluai\.com\.cn/d /yahoo\.cn/d /yahoo\.co\.jp/d /yahoo\.com/d /ja\.wikipedia\.org/d /cloudfront\.net/d' |
                sed '/^[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+$/d' |
                sed '$a bing.com' |
                grep '^[0-9a-zA-Z\.-]\+$' | grep '\.' | sed 's#^\.\+##' | sort -u |
                awk '

**Comment by Jasonzhang2023** [comment]
还有个办法，如果vps服务器是自己搭的，可以到服务器端把分流规则加上geosite:bing 走代理

**Reply by jt8823** [reply]
自建的vps搭建的hy2， 请问如何在服务端添加这条规则啊，  我只配置了最简单的域名伪装！！！看了下官方文档，是不是那个acl功能，  没看明白。

**Reply by Jasonzhang2023** [reply]
如果是hy2，我是用sock5分流给到v2ray，在v2ray里面做的分流。
这里面学问比较多，你自己多看youtube上的视频学习

**Comment by maike2019** [comment]
是的，我也遇到同样的问题，【 4.76-1 】 。在规则列表--代理列表--添加 bing.com 。
在规则管理 --Sing-Box/Xray 分流规则中 添加Bing规则， 内容填写
domain:bing.com
here.com
here.net
heremaps.cn
ovi.com.cn
azure
azure-apim.net
azure-dns.com
azure-dns.info
azure-dns.net
azure-dns.org
azure-mobile.net
azure-sphere.com
azure-test.net
azure.cn
azure.com
azure.net
azurecomcdn.net
azurecontainer.io
azurecosmos.net
azurecosmosdb.com
azurecosmosdb.info
azurecosmosdb.net
azuredatabricks.net
azuredevopslaunch.com
azuredigitaltwin.com
azuredigitaltwins.com
azuredigitaltwins.net
azuredns-prd.info
azuredns-prd.org
azureedge-test.net
azureedge.net
azurefd.net
azurefd.us
azureiotcentral.com
azureiotsolutions.com
azureiotsuite.com
azureplanetscale.info
azureplanetscale.net
azureserviceprofiler.com
azuresmartspaces.net
azurestackvalidation.com
azurewebsites.net
chinacloudapi.cn
chinacloudapp.cn
chinacloudsites.cn
cosmosdb.info
cosmosdb.net
devopsms.com
gotcosmos.com
microsofteca.com
microsoftiotcentral.com
serverlesslibrary.net
signalr.net
tryfunctions.com
windowsazure.cn
windowsazure.com
azure.microsoft.com
azuremarketplace.microsoft.com
bbing.com
bibg.com
biing.com
binb.com
binf.com
bing
bing.com
bing.com.cn
bing.net
bing123.com
bing135.com
bing4.com
bingads.com
bingagencyawards.com
bingapis.com
bingapistatistics.com
binginternal.com
bingit.net
bingiton.com
bingj.com
bingpix.com
bingpk.com
bings.com
bingsandbox.com
bingtoolbar.com
bingtranslator.com
bingvisualsearch.com
bingworld.com
biying.cn
biying.com
biying.com.cn
bluehatnights.com
dictate.ms
discoverbing.com
explorebing.com
flipwithsurface.com
mapblast.com
mappoint.com
masalladeloslimites.com
microsoft-give.com
microsoftcloudsummit.com
microsoftdiplomados.com
microsoftlatamholiday.com
microsoftmxfilantropia.com
microsoftpartnersolutions.com
msnmaps.com
msunlimitedcloudsummit.com
myhomemsn.com
office365love.com
office365tw.com
osdinfra.net
renovacionoffice.com
sprinklesapp.com
working-bing-int.com
bj1.api.bing.com
emoi-cncdn.bing.com
cn.bing.com
cn.bing.net
ditu.live.com
a-msedge.net
c-msedge.net
e-msedge.net
edgesuite.net
evoke-windowsservices-tas.msedge
microsoftedge.com
microsoftedgeinsider.com
msedge.net
s-msedge.net
femalefounderscomp.com
m12.vc
adaptivecards.io
api-extractor.com
apisof.net
appcenter.ms
blazor.net
botframework.com
codethemicrobit.com
devopsassessment.net
dot.net
exp-tas.com
gamesstack.com
graphengine.io
insiderdevtour.com
jwt.ms
microbit.org
microsoftadc.com
microsoftgamestack.com
microsoftiotinsiderlabs.com
microsoftreactor.cn
microsoftreactor.com.cn
microsoftreactor.info
microsoftreactor.net
microsoftreactor.org
microsoftsilverlight.com
microsoftsilverlight.net
microsoftsilverlight.org
microsoftsqlserver.com
mmdnn.com
mono-project.com
msdn.com
msinnovationchallenge.com
mspairlift.com
msropendata.com
nuget.org
nugettest.org
opentranslatorstothings.org
poshtestgallery.com
pwabuilder.com
reactorms.com.cn
sankie.net
sqlserveronlinux.com
timelinestoryteller.com
uwpcommunitytoolkit.com
vfsforgit.com
vfsforgit.org
xamarin.com
developer.microsoft.com
dtlgalleryint.cloudapp.net
poshtestgallery.cloudapp.net
psg-int-centralus.cloudapp.net
psg-int-eastus.cloudapp.net
microsoft-online.cn
microsoft-online.com.cn
microsoftnews.cc
microsoftnews.cn
microsoftnews.com
microsoftnews.net
microsoftnews.org
microsoftnewsforkids.com
microsoftnewsforkids.net
microsoftnewsforkids.org
microsoftnewskids.com
microsoftnewskids.net
microsoftnewskids.org
microsoftol.com
microsoftol.com.cn
microsoftusercontent.com
msn.cn
msn.com
msn.com.cn
msn.com.nsatc.net
msn.com.tw
msnewskids.com
msnewskids.net
msnewskids.org
msnkids.com
winmp.cn
o365cn.com
o365files.cn
o365weve-dev.com
o365weve-ppe.com
o365weve.com
office
office.com
office.net
office365.com
officedev.cn
1drv.com
1drv.ms
livefilestore.com
onedrive.co
onedrive.co.uk
onedrive.com
onedrive.eu
onedrive.live.com
onedrive.net
onedrive.org
storage.live.com
groupme.com
skype.com
skype.net
skypeassets.com
skypeassets.net
skypeforbusiness.com
sway-cdn.com
sway-extensions.com
sway.com
visualstudio-staging.com
visualstudio.co
visualstudio.co.uk
visualstudio.com
visualstudio.eu
visualstudio.net
vsassets.io
vscode-unpkg.net
vscode-webview.net
vscode.dev
live.cn
live.com.akadns.net
sclive.net
windowslive.cn
wlxrs.com
aicscience.com
mmais.com.cn
msminico.cn
msminico.com.cn
msxiaobing.com
msxiaoice.com
msxiaona.cn
officeplus.cn
renlifang.com
cortana.ai
cortanaanalytics.com
cortanaskills.com
forzamotorsport.net
forzaracingchampionship.com
forzarc.com
gamepass.com
msgamestudios.com
orithegame.com
renovacionxboxlive.com
tellmewhygame.com
xbox.co
xbox.com
xbox.eu
xbox.org
xbox360.co
xbox360.com
xbox360.eu
xbox360.org
xboxab.com
xboxgamepass.com
xboxgamestudios.com
xboxlive.cn
xboxlive.com
xboxone.co
xboxone.com
xboxone.eu
xboxplayanywhere.com
xboxservices.com
xboxstudios.com
xbx.lv
callersbane.com
minecraft.net
minecraftshop.com
mojang.com
beth.games
bethesda.net
bethesdagamestudios.com
bethsoft.com
Applicationinsights.io
Applicationinsights.net
aadrm.com
acompli.net
aka.ms
ankarazirvesi2018.com
applicationinsights.io
applicationinsights.net
aria.ms
asp.net
aspnetcdn.com
assets-yammer.com
azurerms.com
binads.com
bluehatil.com
boswp.com
brazilpartneruniversity.com
breakdown.me
centralvalidation.com
ch9.ms
charticulator.com
cloudapp.net
cloudappsecurity.com
crmdynint-gcc.com
crmdynint.com
crossborderexpansion.com
docs.com
dwh5.com
dynamics.com
efproject.net
engkoo.com
fasttrackreadysupport.com
fluidpreview.com
footprintdns.com
gameuxmasterguide.com
gears5.com
gearspop.com
gearstactics.com
getmicrosoftkey.com
gfx.ms
gigjam.com
helpshift.com
hockeyapp.net
hololens.com
hotmail
hotmail.co
hotmail.com
hotmail.eu
hotmail.net
hotmail.org
hummingbird.ms
ie10.com
ie11.com
ie8.co
ie9.com
imaginecup.pl
img-prod-cms-rt-microsoft-com
ingads.com
internetexplorer.co
internetexplorer.com
intunewiki.com
iotinactionevents.com
joinms.com
joinms.com.cn
joinmva.com
kidgrid.tv
kumo.com
latampartneruniversity.com
live.com
live.com.au
live.eu
live.net
livingyourambition.com
localytics.com
lync.com
makecode.org
managedmeetingrooms.com
meetfasttrack.com
meetyourdevices.com
mepn.com
microsoft
microsoft-int.com
microsoft-ppe.com
microsoft-sap-events.com
microsoft-sbs-domains.com
microsoft-smb.cn
microsoft.az
microsoft.be
microsoft.by
microsoft.ca
microsoft.cat
microsoft.ch
microsoft.cl
microsoft.com
microsoft.com.akadns.net
microsoft.com.nsatc.net
microsoft.cz
microsoft.de
microsoft.dk
microsoft.ee
microsoft.es
microsoft.eu
microsoft.fi
microsoft.ge
microsoft.hu
microsoft.io
microsoft.is
microsoft.it
microsoft.jp
microsoft.lt
microsoft.lu
microsoft.lv
microsoft.md
microsoft.net
microsoft.pl
microsoft.pt
microsoft.red
microsoft.ro
microsoft.rs
microsoft.ru
microsoft.se
microsoft.si
microsoft.tv
microsoft.ua
microsoft.uz
microsoft.vn
microsoft365.com
microsoftaccountguard.com
microsoftads.com
microsoftadvertising.com
microsoftadvertisingregionalawards.com
microsoftaffiliates.com
microsoftapps.azurewebsites.net
microsoftazuread-sso.com
microsoftcloud.com
microsoftcloudworkshop.com
microsoftcommunitytraining.com
microsoftemail.com
microsoftestore.com.hk
microsofthouse.com
microsofthouse.net
microsoftinternetsafety.net
microsoftlatamaitour.com
microsoftlinc.com
microsoftonline-p.com
microsoftonline-p.net
microsoftonline.com
microsoftpartnercommunity.com
microsoftready.com
microsoftsiteselection.com
microsoftstore.com
microsoftstore.com.cn
microsoftstore.com.hk
microsoftstream.com
microsoftteams.com
microsofttradein.com
microsofttranslator.com
microsoftuwp.com
momentumms.com
morphcharts.com
mpnevolution.com
ms-studiosmedia.com
ms365surfaceoffer.com
msa.akadns6.net
msads.net
msappproxy.net
msauth.cn
msauth.net
msauthimages.net
mschallenge2018.com
msecnd.net
msft.info
msft.net
msftauth.cn
msftauth.net
msftauthimages.net
msftcenterone.cn
msftconnecttest.com
msfteducation.ca
msftidentity.com
msftnet.org
msgamesresearch.com
msidentity.com
msignitechina.com
msocdn.com
msocsp.com
msopentech.cn
mspil.cn
msra.cn
mstea.ms
msturing.org
msudalosti.com
msvevent.com
mymicrosoft.com
nextechafrica.net
nxta.org
oaspapps.com
oaspapps.com.akadns.net
onecollector.cloudapp.aria
onecollector.cloudapp.aria.akadns.net
onenote.com
onenote.net
onestore.ms
onmicrosoft.com
opticsforthecloud.net
optimizely.com
outingsapp.com
outlook.cn
outlook.com
outlookgroups.ms
outlookmobile.com
passport.net
phonefactor.net
pixapp.net
playfabapi.cn
playfabcn.com
powerappscdn.net
powerautomate.com
powerbi.com
powershellgallery.com
projectmurphy.net
projectsangam.com
pxt.io
s-microsoft.com
s-msft.com
s-msn.com
sfbassets.com
sfbassets.net
sfx.ms
sharepoint.cn
sharepoint.com
sharepointonline.com
skype
staffhub.ms
successwithteams.com
surface.com
svc.ms
syncshop.cn
sysinternals.com
tailwindtraders.com
techhub.ms
tenor.com
tfsallin.net
trafficmanager.net
translatetheweb.com
unlocklimitlesslearning.com
userpxt.io
uservoice.com
videobreakdown.com
videoindexer.ai
virtualearth.net
vsallin.net
wbd.ms
what-fan.net
windows.com
windows.net
windows-int.net
windows-ppe.net
windows.com
windows.net
windows.nl
windows8.hk
windowscommunity.net
windowsmarketplace.com
windowsphone-int.com
windowsphone.com
windowssearch.com
windowsupdate.com
winhec.com
winhec.net
yammer.com
yammerusercontent.com
g.akamaiedge.net
a1158.g.akamai.net
a122.dscg3.akamai.net
a767.dscg3.akamai.net
cs11.wpc.v0cdn.net
cs9.wac.phicdn.net
b.akamaiedge.net
spoprod-a.akamaihd.net
windows.com.akadns.net
prod-video-cms-rt-microsoft-com.akamaized.net
statics-marketingsites-eas-ms-com.akamaized.net
statics-marketingsites-neu-ms-com.akamaized.net
vip5.afdorigin-prod-am02.afdogw.com
img-prod-cms-rt-microsoft-com.akamaized.net
img-s-msn-com.akamaized.net
mwf-service.akamaized.net
officecdn-microsoft-com.akamaized.net
statics-marketingsites-eus-ms-com.akamaized.net
statics-marketingsites-wcus-ms-com.akamaized.net
同样没有效果，并且我已经把给规则放在第三个位置了，第一个是AD去广告，第二个是BT 黑洞。

**Reply by gito10** [reply]
你的代理模式，就是那几个代理列表，勾选了哪些，或者是点的下面的模式自动勾选的。如果该勾的列表都勾了，还有一个可能就是你的DNS是不是配置不当，走国内请求了，这样拿到的就是国内IP。再就是可以换几个节点试试。还有检查下直连列表里面是不是有 **bing.com**，反正里面是有**microsoft.com**，不过这些都不影响 Bing，我的就只添加了 **bing.com**，完全没问题。

**Reply by gito10** [reply]
你可以把 **ip111.cn**加入代理列表和某个分流规则，然后访问 http://ip111.cn 看看国内IP和谷歌IP 是不是对应节点的出口IP。

**Comment by q2051839** [comment]
直接在屏蔽列表里屏蔽cn.bing.com不就行了。这么复杂么。

**Reply by maike2019** [reply]
尝试了你说的在passwall 规则列表--屏蔽列表中 加入  cn.bing.com ，用passwall访问 bing.com，依然是国内bing，无法开启AI聊天 Copilot

**Reply by q2051839** [reply]
可能我在DNS流程，使用了adg，同时屏蔽了cn.bing.com。我不确定没有其他的改动。正常使用ai

**Comment by gito10** [comment]
如果把**bing.com**手动加入代理列表，访问 Bing 还是跳转 cn.bing.com，很大可能是没有开启“过滤代理域名IPv6”（或开启了但没生效，下面详说），又没开启IPv6的透明代理（大部分节点也不支持IPv6），然后浏览器请求到IP，优先选择了IPv6的地址来访问服务器，所以是直连，就跳转cn子域名了。
![image](https://github.com/xiaorouji/openwrt-passwall/assets/10704839/a93789e2-97b9-4944-a35f-96dd5047dfac)

如果上图所示开关未开，可以确定就是这个原因。如果开关是开的，需要进一步确认是否生效，也就是看浏览器访问 Bing时，到底是用的 IPv4地址，还是 IPv6地址。
可以`nslookup bing.com`，查看结果是否有IPv6地址。

目前我遇到的 **过滤代理域名IPv6**开关失效的情况:
- Xray分流模式，开启 ChinaDNS-NG，IPv6过滤失效，ChinaDNS-NG 配置检查没问题，~~还在排查中。~~ 

问题原因已找到。
使用分流模式时，会为分流规则中的走代理的域名生成 dnsmasq的server记录，直接指定 **#15353**端口解析，例如 `server=/.bing.com/127.0.0.1#15353`，越过ChinaDNS-NG了，所以不能过滤AAAA记录。

**Reply by gito10** [reply]
@lwb1978 上面我发现的这个问题，现在应该已经解决了吧？解决了我就把帖删了。

**Reply by lwb1978** [reply]
不是很清楚，这个时间我还没开始加入维护队伍。

**Reply by gito10** [reply]
那只能实测一下看看了，先留着吧。

**Comment by ghost** [comment]
還有一種可能是：代理出口所在的IP段被Microsoft認爲是中國了

已經知道的事情是：multacom和colocrossing的位址會這樣
